### PR TITLE
per bugfix #21824 pass the message which cause restart in PersistentActor

### PR DESCRIPTION
**Pull request by naferx**

### CODE REVIEW

**Comment by naferx** [comment]
Refs https://github.com/akka/akka/issues/21824

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala [hunk 1]
```diff
@@ -205,9 +205,9 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
         case Some(ReplayedMessage(m)) ⇒
           flushJournalBatch()
           super.aroundPreRestart(reason, Some(m))
-        case mo ⇒
+        case mo: Option[Any] ⇒
```
Comment (drewhk) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: why the explicit type?

**Review by drewhk — APPROVED** [review]
LGTM

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala [hunk 2]
```diff
@@ -205,9 +205,9 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
...
           flushJournalBatch()
-          super.aroundPreRestart(reason, None)
+          super.aroundPreRestart(reason, mo)
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: 👍

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala (hunk 1 above)
Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: that should not be needed

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala [hunk 3]
```diff
@@ -207,7 +207,7 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
           super.aroundPreRestart(reason, Some(m))
         case mo ⇒
           flushJournalBatch()
-          super.aroundPreRestart(reason, None)
+          super.aroundPreRestart(reason, mo)
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: @drewhk @naferx @patriknw 
I removed the type match from this clause, so this PR is ready to merge.

Comment (naferx) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: @drewhk @patriknw 
Sorry for the delayed response. I forgot to remove it.
Thanks! @ktoso

Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: No problem at all, thanks for the PR!

**Pull request merged**